src/moudule/crud.js

```diff
@@ -1,46 +1,144 @@
-
-  const checkboxContainer = document.createElement('div');
-  checkboxContainer.classList.add('checkboxContainer');
-  const checkbox = document.createElement('input');
-  checkbox.type = 'checkbox';
-  checkbox.id = `check${task.index}`;
-
-  label.htmlFor = `check${task.index}`;
-  label.innerText = task.description;
-
-  label.addEventListener('click', () => {
-    isSelected = !isSelected;
-    if (isSelected === true) {
-      label.style.textDecoration = 'line-through';
-    } else {
-      label.style.textDecoration = 'none';
-    }
-  });
-
-  checkbox.addEventListener('click', () => {
-    if (checkbox.checked) {
-      label.style.textDecoration = 'line-through';
-    } else {
-      label.style.textDecoration = 'none';
-    }
-  });
-
-  checkboxContainer.append(checkbox, label);
-  itemTaskElem.append(checkboxContainer, iconTask);
-  li.appendChild(itemTaskElem);
-  return li;
-}
-
-export default Crud;
\ No newline at end of file
+const containerElement = document.querySelector('.containerTodo');
+const ulElement = document.querySelector('.ulElement');
+
+const Task = class {
+  constructor(description, completed = false, index = 1) {
+    this.description = description;
+    this.completed = completed;
+    this.index = index;
+    this.tasks = [
+      {
+        description: 'wash the dogs',
+        completed: false,
+        index: 0,
+      },
+      {
+        description: 'Complete To Do list project',
+        completed: false,
+        index: 1,
+      },
+    ];
+  }
+
+  populateField = () => {
+    let i = -1;
+    this.tasks.forEach((task) => {
+      i += 1;
+      task.index = i;
+    });
+    localStorage.setItem('savedTasks', JSON.stringify(this.tasks));
+  };
+
+  addTask = (newTask) => {
+    this.tasks.push(newTask);
+    this.populateField();
+    this.showTasks();
+  };
+
+  removeTask(task) {
+    const result = this.tasks.filter((b) => b !== task);
+    this.tasks = result;
+    this.populateField();
+  }
+
+  showTasks = () => {
+    ulElement.innerHTML = '';
+    this.tasks.map((task) => {
+      const li = document.createElement('li');
+      const itemElement = document.createElement('div');
+      const checkbox = document.createElement('input');
+      const checkboxContainer = document.createElement('div');
+      checkboxContainer.classList.add('checkboxContainer');
+      const label = document.createElement('label');
+      const icon = document.createElement('div');
+      const textInput = document.createElement('input');
+      textInput.classList.add('textInput', 'hidden');
+      icon.classList.add('iconContainer');
+      itemElement.classList.add('itemElement');
+      icon.innerHTML = '<i class="fas fa-ellipsis-v"></i>';
+      const iconMenu = '<i class="fas fa-ellipsis-v"></i>';
+      const iconDelete = '<i class="fas fa-trash-alt"></i>';
+      textInput.value = task.description;
+      checkbox.type = 'checkbox';
+      checkbox.id = `check${task.index}`;
```

Comment:
I suggest you refactor this so that, the items can have indexes corresponding to their position. As per the [project requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md), a project with `index 1` should be the `first` item
<img width="1446" alt="Screenshot 2023-01-19 at 10 49 36" src="https://user-images.githubusercontent.com/48546992/213384782-ce679759-acdc-4749-9ae6-5daf8450bb32.png">
This is not the case with your items:
<img width="1674" alt="Screenshot 2023-01-19 at 10 47 25" src="https://user-images.githubusercontent.com/48546992/213385165-ddf17fe1-baed-456e-b3f3-9fae25bbee72.png">